src/commonMain/kotlin/io/ktor/io/ByteArrayBufferPool.kt

```diff
@@ -21,3 +21,26 @@ public class ByteArrayBufferPool(
         }
     }
 }
+
+public class ByteArrayPool(
+    private val size: Int = DEFAULT_BUFFER_SIZE,
+    capacity: Int = DEFAULT_POOL_CAPACITY,
+) : DefaultPool<ByteArray>(capacity) {
+
+    override fun produceInstance(): ByteArray = ByteArray(size)
+
+    override fun clearInstance(instance: ByteArray): ByteArray {
+        instance.fill(0)
+        return instance
+    }
+
+    public companion object {
+        public val Default: ObjectPool<ByteArray> = ByteArrayPool()
+
+        public val NoPool: NoPoolImpl<ByteArray> = object : NoPoolImpl<ByteArray>() {
```

Comment:
Looks like `NoPool` implementation can be single per all pools with `NoPoolImpl<Nothing>` and cast, no?
similar to `EmptyList`

Reply:
Let's unify it after the design phase